cmd/vic-machine/common/limits.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package common
+
+import (
+	"gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/govmomi/vim25/types"
+	"github.com/vmware/vic/pkg/flags"
+)
+
+type ResourceLimits struct {
+	VCHCPULimitsMHz       *int              `cmd:"cpu"`
```

Comment:
The problem is to differentiate default value and no input at all. With nil value of pointer int, we clearly know if user has that option or not, and no requirement for end user.
The downside of this option is how to show the default value. nil is not likely a good one.

But with general int value, we do not have a way to know if user has default value as input or no input at all. So the way is to define a special default value. If we find that value, we "think" the value is not from user input.
In this way, if no input for this parameter, cli will set the invalid value as option value. So our code has to accommodate the invalid value to make it work, e.g. default value for VCHCPUReservationMHz is -1. If we found that's -1, we need to change that value to 0, and move forward without any error message.
if user do have -1 as input, we'll move forward in the same way, even that's not a valid input.

So in this way, if customer used some special value, they'll fall into the black hole, without any useful message.

atm, I still prefer to keep optional int solution. But it's always good to have such discussion.